test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8411,108 +8436,102 @@ def is_history_enabled(self):
             return True
         return False
 
-    def cleanup_jobs(self, extend=None, runas=None):
+    def cleanup_jobs(self):
         """
         Helper function to delete all jobs.
         By default this method will determine whether
         job_history_enable is on and will cleanup all history
         jobs. Specifying an extend parameter could override
         this behavior.
-
-        :param runas: Clean the job as
-        :type runas: str or None
         """
-        delete_xt = 'force'
+        delete_xt = 'nomailforce'
         select_xt = None
         if self.is_history_enabled():
             delete_xt += 'deletehist'
             select_xt = 'x'
-        job_ids = self.select(extend=select_xt)
+        jobs = self.status(JOB, extend=select_xt)
+        job_ids = sorted(list(set([x['id'] for x in jobs])))
+        host_pid_map = {}
+        for job in jobs:
+            if 'exec_host2' in job:
+                exec_host = job['exec_host2']
+            else:
+                exec_host = job.get('exec_host', None)
+            if not exec_host or 'session_id' not in job:
+                continue
+            _host = exec_host.split('/')[0].split(':')[0]
+            if _host not in host_pid_map:
+                host_pid_map.setdefault(_host, [])
+            host_pid_map[_host].append(job['session_id'])
 
         # Turn off scheduling so jobs don't start when trying to
         # delete. Restore the orignial scheduling state
         # once jobs are deleted.
-        sched_state = {}
-        for sc in self.schedulers:
-            sched_state[sc] = self.status(
-                SCHED, {'scheduling'},
-                id=self.schedulers[sc].sc_name)[0]['scheduling']
-            # runas is required here because some tests remove current user
-            # from managers list
-            self.manager(MGR_CMD_SET, SCHED, {'scheduling': 'False'},
-                         id=self.schedulers[sc].sc_name, runas='root')
-        num_jobs = len(job_ids)
-        if num_jobs >= 100:
-            self._cleanup_large_num_jobs(job_ids, runas=runas)
-
-        if num_jobs > 0 and num_jobs < 100:
+        sched_state = []
+        scheds = self.status(SCHED)
+        for sc in scheds:
+            if sc['scheduling'] == 'True':
+                sched_state.append(sc['id'])
+                # runas is required here because some tests remove
+                # current user from managers list
+                a = {'scheduling': 'False'}
+                self.manager(MGR_CMD_SET, SCHED, a, id=sc['id'],
+                             runas='root')
+                self.expect(SCHED, a, id=sc['id'])
+        try:
+            self.deljob(id=job_ids, extend=delete_xt, runas='root', wait=False)
```

Comment:
If we do kill session-id then the server will get obit storm, also here we are doing qdel with Wforce as a root user so the server will not contact to mom at all.

Reply:
I don't think using Wforce option means that the server will not contact mom at all. It means that the server will send a deletejob request to mom and then delete jobs without waiting for an Obit from mom. 
I see your point that killing session ids might trigger an obit storm and it could potentially be slower.

Reply:
@arungrover Sorry my mistake I didn't mean "contact" but I mean "wait for mom's reply on delete job"